### PR TITLE
supplemental updates to cf-deployment for loggregator-agent 6.0.0

### PR DESCRIPTION
**Pull request by Benjamintf1**
### WHAT is this change about?

- ~update loggregator-agent to 6.0.0~ this bump has already been made to develop
- include ops file to turn off logs in firehose
- update logcache syslog ingress to still include metrics

### What customer problem is being addressed? Use customer persona to define the problem e.g. Alana is unable to...

Operators can now turn off logs to firehose, allowing them to scale down the doppler and traffic controller/rlp components

### Has a cf-deployment including this change passed [cf-acceptance-tests](https://github.com/cloudfoundry/cf-acceptance-tests)?

- [x] YES
- [ ] NO

### Does this PR introduce a breaking change? Please take a moment to read through the examples before answering the question.

- [ ] YES - please choose the category from below. Feel free to provide additional details.
- [x] NO

### How should this change be described in cf-deployment release notes?

> _Something brief that conveys the change and is written with the **persona (Alana, Cody...)** in mind. See [previous release notes](https://github.com/cloudfoundry/cf-deployment/releases) for examples._

### Does this PR introduce a new BOSH release into the base cf-deployment.yml manifest or any ops-files?

- [ ] YES - please specify
- [x] NO

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!

@pianohacker @MasslessParticle @JSchuenke

### CODE REVIEW

**Comment by pianohacker** [comment]
Note; while this is a change in behavior, we do not expect this to be a breaking change for most consumers. The consumers we've talked to so far explicitly do NOT want metrics. Nevertheless, it's worth noting the following in the release notes:

- Aggregate syslog drains (as configured by the `aggregate_drains` property on the `loggregator_agent` and `loggregator_agent_windows`) now send only logs by default; users can opt into receiving metrics by adding the `?include-metrics-deprecated=true` flag to their aggregate drain URLs.
- Users that are sending application logs via other means (such as syslog drains) can now opt out of logs being transmitted in the firehose via the new `disable-logs-in-firehose.yml`. To preserve log availability in the CF CLI, the `enable-syslog-agent.yml` and `enable-logcache-syslog-ingress.yml` ops files should also be used.

**Comment by davewalter** [comment]
Hi @Benjamintf1,

Our [CI](https://release-integration.ci.cf-app.com/teams/main/pipelines/update-releases/jobs/update-loggregator-agent/builds/55) already automatically validated and consumed v6.0.0 of loggregator-agent. Could you please remove the version bump from this PR?

Also, does the new release introduce the breaking change, or is it the ops-files that are breaking? I believe that we are scheduled to work on v13 of cf-deployment next week, so now might be the perfect time for this, but @Syerram might be concerned that this change was not included in the RFC scope doc he sent to the community. I will leave it to him to decide whether this can be included in v13 or not.

Regards,
Dave

**Comment by davewalter** [comment]
Hi @Benjamintf1,

We are reviewing this in IPM for inclusion in v13, and we noticed that there are changes for Windows1803 in the PR. We are actually going to remove all references to Windows 1803 as part of v13, so could you please remove those parts of this PR, in addition to the version bump I mentioned earlier?

Thanks,
Dave

**Comment by Benjamintf1** [comment]
So if you update to 6.0.0 of loggregator-agent and are using the syslog-egress to log-cache(or aggregate drains at all), you will no longer see metrics in the stream. This is the big breaking change, although we strongly suspect that only the log-cache change(which should be fixed by the pr) would actually be considered breaking in the sense of the larger cf product.

**Comment by jamespollard8** [comment]
@Benjamintf1 a couple small remaining things:
1) Could you please provide us a sentence (or 2) to include in the cf-d release notes?
2) Could you please provide us with AC so that our PM can validate the change?

**Comment by davewalter** [comment]
Hi @Benjamintf1,

Could you also please resolve the merge conflict from us merging #867?

Thanks,
Dave

**Comment by JSchuenke** [comment]
@jamespollard8 Would these notes by Jesse work? https://github.com/cloudfoundry/cf-deployment/pull/879#issuecomment-619161691

**Comment by jamespollard8** [comment]
> @jamespollard8 Would these notes by Jesse work?

Yeah, I did see those. They're a little longer than average for release notes but we'll make do 😁 

>Happy to give some specific AC if necessary!

Please do. We'll likely be ready to pick this up today.

Thanks!

**Comment by JSchuenke** [comment]
> @jamespollard8
Sweet! Here's the AC for the two features we added:

**BREAKING CHANGE: Aggregate syslog drains defaulting to no logs**

Given a PAS environment
When I set up a syslog:// or syslog-tls:// aggregate drain
Then I see only logs at its destination, not metrics

When I enable Log Cache syslog ingestion and look at Log Cache
Then I still see component metrics

When I set up an app syslog drain with metrics enabled
Then I still see component metrics

**Opting out of logs in the logstream**

Given a CF with apps/components emitting logs and metrics
When I redeploy with the new metric_envelopes_only property (which defaults to false) enabled on the loggregator_agent job
Then I see no log or event envelopes in cf log-stream

**Comment by jamespollard8** [comment]
More detailed AC instructions/plan:

**"before" testing** 
deploy from current develop using `operations/experimental/use-logcache-syslog-ingress.yml`,
appending to `  value: "syslog-tls://doppler.service.cf.internal:6067?include-metrics-deprecated=true"`, make it `  value: "syslog-tls://doppler.service.cf.internal:6067, syslog://logs6.papertrailapp.com:34537"

"after" testing:
do the exact same thing except deploy from this branch instead of develop and using nearly the identical ops file and custom appending of aggregate drain making it:
`  value: "syslog-tls://doppler.service.cf.internal:6067?include-metrics-deprecated=true, syslog://logs6.papertrailapp.com:34537"
*and* this ops file: `operations/experimental/disable-logs-in-firehose.yml`



"before" test for
> Opting out of logs in the logstream

> Given a CF with apps/components emitting logs and metrics
> When I redeploy with the new metric_envelopes_only property (which defaults to false) enabled on the loggregator_agent job
> Then I see no log or event envelopes in cf log-strea




after test:
deploy from branch using

File: operations/experimental/disable-logs-in-firehose-windows2019.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+- type: replace
+  path: /instance_groups/name=windows2019-cell/jobs/name=loggregator_agent_windows/properties/disable_logs?
+  value: true
+
+- type: replace
+  path: /instance_groups/name=windows2019-cell/jobs/name=loggregator_agent_windows/properties/disable_logs?
```
Comment (jamespollard8) [inline operations/experimental/disable-logs-in-firehose-windows2019.yml:7]: this appears to be exactly the same as the block above

File: operations/experimental/disable-logs-in-firehose.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+- type: replace
+  path: /addons/name=loggregator_agent/jobs/name=loggregator_agent/properties/disable_logs?
+  value: true
+
+- type: replace
+  path: /addons/name=loggregator_agent/jobs/name=loggregator_agent/properties/disable_logs?
```
Comment (jamespollard8) [inline operations/experimental/disable-logs-in-firehose.yml:7]: this appears to be exactly the same as the block above

File: operations/experimental/disable-logs-in-firehose-windows2019.yml (hunk 1 above)
Comment (Benjamintf1) [reply operations/experimental/disable-logs-in-firehose-windows2019.yml:7]: honestly, thinking about it right now, I'm not sure how that happened.

File: operations/experimental/disable-logs-in-firehose.yml (hunk 2 above)
Comment (Benjamintf1) [reply operations/experimental/disable-logs-in-firehose.yml:7]: indeed, that might be a mistake.

**Comment by jamespollard8** [comment]
@Benjamintf1 @JSchuenke heads up, there's another minor merge conflict (from removed windows ops files) that you'll need to resolve.

**Comment by jamespollard8** [comment]
In the before test (not including `operations/experimental/disable-logs-in-firehose.yml`) we do see logs come through
from running `$ cf log-stream $(cf app my-app --guid)`

metric line: `…"gauge":{"metrics":{"cpu":{"unit":"percentage","value":0.27095937069274084},…`
log line: `"method":"GET","origin":"gorouter","peer_type":"Client","remote_address":"1.2.3.4"`

in the after test with the disable-logs-in-firehose and updated `operations/experimental/use-logcache-syslog-ingress.yml`, we [as expected]:
1) no longer see logs or events through `cf log-stream $(cf app my-app-new --guid) -t log` or `cf log-stream $(cf app my-app-new --guid) -t event`
and
2) we do see metrics flowing when we run `cf tail -c metrics my-app-new -f`

Accepting

**Pull request merged**